lib/jekyll/cache.rb

```diff
@@ -0,0 +1,175 @@
+# frozen_string_literal: true
+
+require "digest"
+require "fileutils"
+require "pstore"
+
+module Jekyll
+  class Cache
+    extend Forwardable
+
+    # rubocop:disable Style/ClassVars
+    @@caches = {}
+    @@disk_cache_enabled = true
+
+    # Get an existing named cache, or create a new one if none exists
+    #
+    # name - name of the cache
+    #
+    # Returns nothing.
+    def initialize(name)
+      @@base_dir ||= File.expand_path(".jekyll-cache/Jekyll/Cache")
```

Comment:
This is the only query that remains unanswered..

Reply:
What we need to do is make this relative to `site.source` instead of `pwd`.
I’m not sure of the syntactically correct way to do that.

I agree this does not belong here.

Reply:
#7160 once merged, will give you the helper method to call a path within `.jekyll-cache` relative to `site.source`

Reply:
The problem is that we need to get an instance of `site`. The helper method you added is not static.

I hate singletons.

Reply:
I didn't realize that this class doesn't link to a particular Site instance.
But we do have access to the `config` hash...
I guess, the issue will have to be handled via a separate PR after **this PR** is merged to `master`.